File: backend/src/main/java/com/zenith/backend/service/CompanyService.java
package com.zenith.backend.service;

import com.zenith.backend.model.Conversation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class CompanyService {

    private final OpenRouterService openRouterService;

    private static final List<String> COMPANIES = List.of(
        "Amazon", "Google", "Microsoft", "Adobe", "Oracle", "Goldman Sachs",
        "JPMorgan", "Flipkart", "Walmart", "Uber", "Atlassian", "NVIDIA",
        "Qualcomm", "Intel", "Apple", "Meta", "Netflix", "TCS", "Infosys",
        "Wipro", "Accenture", "Cognizant", "Capgemini", "Deloitte"
    );

    @Autowired
    public CompanyService(OpenRouterService openRouterService) {
        this.openRouterService = openRouterService;
    }

    public String extractCompany(String query) {
        String lowerQuery = query.toLowerCase();
        for (String company : COMPANIES) {
            if (lowerQuery.contains(company.toLowerCase())) {
                return company;
            }
        }
        return "General";
    }

    public String getCompanyInsights(String query, Conversation conversation) {
        String company = extractCompany(query);
        if (!"General".equals(company)) {
            conversation.setActiveCompany(company);
        } else if (conversation.getActiveCompany() != null) {
            company = conversation.getActiveCompany();
        }

        String systemPrompt = "You are Zenith AI's Company Intelligence Engine. You are an expert placement mentor.";
        String userPrompt = String.format("""
                Return the response in clean GitHub Markdown.
                
                Rules:
                - Use ## for section headings.
                - Use bullet points wherever possible.
                - Use tables only if they improve readability.
                - Do NOT add unnecessary blank lines.
                - Do NOT use excessive separators (---).
                - Keep spacing compact.
                - Be concise but complete.
                - Use only the sections relevant to the user's query.
                
                Company:
                %s
                
                User Query:
                %s
                
                Possible sections:
                - Hiring Process
                - Interview Rounds
                - Online Assessment (OA)
                - Important Topics
                - Skills Required
                - Expected Projects
                - Salary (Approx.)
                - Preparation Tips
                
                If the user asks about only one topic (e.g. salary or interview process), answer only that topic instead of all sections.
                """, company, query);

        return openRouterService.ask(systemPrompt, userPrompt);
    }
}

File: backend-python/services/company_service.py
from services.general_service import ask_zenith
from utils.session import conversation_state
import re



COMPANIES = [
    "Amazon",
    "Google",
    "Microsoft",
    "Adobe",
    "Oracle",
    "Goldman Sachs",
    "JPMorgan",
    "Flipkart",
    "Walmart",
    "Uber",
    "Atlassian",
    "NVIDIA",
    "Qualcomm",
    "Intel",
    "Apple",
    "Meta",
    "Netflix",
    "TCS",
    "Infosys",
    "Wipro",
    "Accenture",
    "Cognizant",
    "Capgemini",
    "Deloitte"
]


def extract_company(query):

    query_lower = query.lower()

    for company in COMPANIES:

        if company.lower() in query_lower:

            return company

    return "General"


def company_insights(query):

    company = extract_company(query)

    if company != "General":
        conversation_state["company"] = company
    else:
        company = conversation_state.get("company", "General")

    prompt = f"""
You are Zenith AI's Company Intelligence Engine.

Return the response in clean GitHub Markdown.

Rules:
- Use ## for section headings.
- Use bullet points wherever possible.
- Use tables only if they improve readability.
- Do NOT add unnecessary blank lines.
- Do NOT use excessive separators (---).
- Keep spacing compact like ChatGPT.
- Be concise but complete.
- Use only the sections relevant to the user's query.

Company:
{company}

User Query:
{query}

Possible sections:
- Hiring Process
- Interview Rounds
- Online Assessment (OA)
- Important Topics
- Skills Required
- Expected Projects
- Salary (Approx.)
- Preparation Tips

If the user asks about only one topic (e.g. salary or interview process), answer only that topic instead of all sections.
"""

    response= ask_zenith(
        "You are an expert placement mentor.",
        prompt
    )

    response = re.sub(r"\n{3,}", "\n\n", response)
    conversation_state["history"].append(
        {"user": query, "assistant": response}
    )
    conversation_state["history"] = conversation_state["history"][-5:]

    return response